From 108565f898a04d1eac815747133679eed50c5865 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Wed, 25 Jan 2023 10:58:45 +0100
Subject: [PATCH 11/27] ENH: Add variable install dirs for Libs dev files

This removes the hardcoded `include` component of the installation path
for the installation of development files. This affects
`ITKFactoryRegistration`, `MRML`, `RemoteIO`, `vtkITK`,
`vtkSegmentationCore`, `vtkTeem`.
---
 Libs/ITKFactoryRegistration/CMakeLists.txt  | 2 +-
 Libs/MRML/CLI/CMakeLists.txt                | 2 +-
 Libs/MRML/Core/CMakeLists.txt               | 2 +-
 Libs/MRML/DisplayableManager/CMakeLists.txt | 2 +-
 Libs/MRML/IDImageIO/CMakeLists.txt          | 2 +-
 Libs/MRML/Logic/CMakeLists.txt              | 2 +-
 Libs/MRML/Widgets/CMakeLists.txt            | 2 +-
 Libs/RemoteIO/CMakeLists.txt                | 5 +++--
 Libs/vtkITK/CMakeLists.txt                  | 2 +-
 Libs/vtkSegmentationCore/CMakeLists.txt     | 2 +-
 Libs/vtkTeem/CMakeLists.txt                 | 2 +-
 11 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/Libs/ITKFactoryRegistration/CMakeLists.txt b/Libs/ITKFactoryRegistration/CMakeLists.txt
index 76875294e7..6f5c27bea9 100644
--- a/Libs/ITKFactoryRegistration/CMakeLists.txt
+++ b/Libs/ITKFactoryRegistration/CMakeLists.txt
@@ -75,7 +75,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.(h|txx)")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
diff --git a/Libs/MRML/CLI/CMakeLists.txt b/Libs/MRML/CLI/CMakeLists.txt
index e18d76231f..c70e66bb7e 100644
--- a/Libs/MRML/CLI/CMakeLists.txt
+++ b/Libs/MRML/CLI/CMakeLists.txt
@@ -66,7 +66,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
diff --git a/Libs/MRML/Core/CMakeLists.txt b/Libs/MRML/Core/CMakeLists.txt
index 45b69783e4..c9bcb281a7 100644
--- a/Libs/MRML/Core/CMakeLists.txt
+++ b/Libs/MRML/Core/CMakeLists.txt
@@ -127,7 +127,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
diff --git a/Libs/MRML/DisplayableManager/CMakeLists.txt b/Libs/MRML/DisplayableManager/CMakeLists.txt
index 625eab5bb9..1b5d8aab12 100644
--- a/Libs/MRML/DisplayableManager/CMakeLists.txt
+++ b/Libs/MRML/DisplayableManager/CMakeLists.txt
@@ -62,7 +62,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
diff --git a/Libs/MRML/IDImageIO/CMakeLists.txt b/Libs/MRML/IDImageIO/CMakeLists.txt
index 718c9b6a55..848f562fe1 100644
--- a/Libs/MRML/IDImageIO/CMakeLists.txt
+++ b/Libs/MRML/IDImageIO/CMakeLists.txt
@@ -70,7 +70,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
diff --git a/Libs/MRML/Logic/CMakeLists.txt b/Libs/MRML/Logic/CMakeLists.txt
index 1a28d303d5..d9f47ac67b 100644
--- a/Libs/MRML/Logic/CMakeLists.txt
+++ b/Libs/MRML/Logic/CMakeLists.txt
@@ -55,7 +55,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
diff --git a/Libs/MRML/Widgets/CMakeLists.txt b/Libs/MRML/Widgets/CMakeLists.txt
index 53d82dce39..0232af47cd 100644
--- a/Libs/MRML/Widgets/CMakeLists.txt
+++ b/Libs/MRML/Widgets/CMakeLists.txt
@@ -433,7 +433,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
   install(
     FILES ${headers}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
   # --------------------------------------------------------------------------
diff --git a/Libs/RemoteIO/CMakeLists.txt b/Libs/RemoteIO/CMakeLists.txt
index 46a48e7127..43c8dffb0d 100644
--- a/Libs/RemoteIO/CMakeLists.txt
+++ b/Libs/RemoteIO/CMakeLists.txt
@@ -65,7 +65,8 @@ configure_file(
   )
 
 # --------------------------------------------------------------------------
-# Install headers
+#
+#Install headers
 # --------------------------------------------------------------------------
 if(NOT DEFINED ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   set(${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT ON)
@@ -74,7 +75,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
diff --git a/Libs/vtkITK/CMakeLists.txt b/Libs/vtkITK/CMakeLists.txt
index f451099f8b..3e0387ed64 100644
--- a/Libs/vtkITK/CMakeLists.txt
+++ b/Libs/vtkITK/CMakeLists.txt
@@ -99,7 +99,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.(h|txx)")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
diff --git a/Libs/vtkSegmentationCore/CMakeLists.txt b/Libs/vtkSegmentationCore/CMakeLists.txt
index 4b11ca3692..4873578b55 100644
--- a/Libs/vtkSegmentationCore/CMakeLists.txt
+++ b/Libs/vtkSegmentationCore/CMakeLists.txt
@@ -23,7 +23,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.(h|txx)")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
diff --git a/Libs/vtkTeem/CMakeLists.txt b/Libs/vtkTeem/CMakeLists.txt
index a30ea518b3..c8ac7a7ef9 100644
--- a/Libs/vtkTeem/CMakeLists.txt
+++ b/Libs/vtkTeem/CMakeLists.txt
@@ -64,7 +64,7 @@ if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
   file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.(h|txx)")
   install(
     FILES ${headers} ${CMAKE_CURRENT_BINARY_DIR}/${configure_header_file}
-    DESTINATION include/${PROJECT_NAME} COMPONENT Development)
+    DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR} COMPONENT Development)
 endif()
 
 # --------------------------------------------------------------------------
-- 
2.39.2

